### DISCUSSION TITLE
MIP-1 wallet_watchAsset for NFTs

### DISCUSSION THREAD

**Discussion by vandan** [MetaMask/metamask-improvement-proposals#22] [MIPs]
[MIP-1](https://github.com/MetaMask/metamask-improvement-proposals/blob/main/MIPs/mip-1.md), which introduces an NFT extension for wallet_watchAsset, is available in the MetaMask Extension. 
Please read the [HowTo for "Displaying NFTs"](https://docs.metamask.io/wallet/how-to/register-token/#display-nfts) (ERC-721 & ERC-1155 tokens) and give it a try using the [interactive API reference for wallet_watchAsset](https://docs.metamask.io/wallet/reference/wallet_watchasset/).

While use of this API with `type` set to ERC721 or ERC1155 is considered experimental and subject to change, we are interested in your experiences with the new method.

Please share your feedback and concerns here.

**Comment by SCBuergel** [comment]
Cool proposal to finally support NFTs. You did mention the IP privacy risks in your security section. While we at HOPR are working towards a much better solution than a trusted proxy, we developed this [non-private NFT](https://mint.hoprnet.org) to highlight not just the IP but also other user metadata privacy issues with NFTs.

So beyond the IP address, it would be great if developers would watch out to make the request that fetches the image as privacy respecting as possible and not send e.g. user metadata in the header.

**Reply by vandan** [reply]
We definitely agree with the sentiment. The non-private NFT link reference doesn't seem to be working though. Can you re-share so that we can bring this to the attention of MetaMask folks who work on NFTs/security?

**Reply by SCBuergel** [reply]
Updated to point to https://mint.hoprnet.org
